hooks/build

```
@@ -0,0 +1,307 @@
+#!/usr/bin/env bash
+
+#
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+
+# This is hook build used by DockerHub. We are also using it
+# on Travis CI to potentially rebuild (and refresh layers that
+# are not cached) Docker images that are used to run CI jobs
+
+set -euo pipefail
+
+MY_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+date
+
+echo
+echo "Build hook is running."
+echo "IMAGE_NAME=${IMAGE_NAME:=}"
+echo "CACHE_TAG=${CACHE_TAG:=}"
+echo
+
+# Remove index.docker.io/ prefix as it is added by default by DockerHub
+export LOCAL_IMAGE_NAME=${IMAGE_NAME#index.docker.io/}
+
+echo
+echo "LOCAL_IMAGE_NAME=${LOCAL_IMAGE_NAME}"
+echo
+
+cd ${MY_DIR}/..
+
+# You can set SKIP_CACHE to true if you do not want to use Docker cache during build
+# This way you can be sure that building docker from scratch will work fine
+SKIP_CACHE=${SKIP_CACHE:=false}
+
+AIRFLOW_VERSION=$(cat airflow/version.py | grep version | awk {'print $3'} | tr -d "'+")
```

```suggestion
AIRFLOW_VERSION=$(python -c "from airflow.version import version; print(version)")
```
What do you think?  This can be a more stable solution than reading source codes. as text files.

---

This requires importing airflow, which will pull in airflow configuration which will write default config files of thru don't exist :(

---

Yeah. That's exactly why I chose cat/grep/awk as a simple hack to avoid any dependencies than the posix one that you can 100% rely on :)

---

One option might be to follow https://packaging.python.org/guides/single-sourcing-package-version/ and put this login in setup.py, which can then be run as `python ./setup.py version` without importing airflow code.

---

OK. Will try to make something less POSIXy and mory Pythony :)

---

I will keep it open and try to make more pythony solution as mentioned above.